components/post_view/post_image.jsx

```diff
@@ -99,14 +103,10 @@ export default class PostImageEmbed extends React.PureComponent {
 
     render() {
         if (this.state.errored || !this.state.loaded) {
+            // scroll pop could be improved with better placeholder during !this.  state.loading
             return null;
         }
-
-        var url = this.props.link;
-        if (global.window.mm_config.HasImageProxy === 'true') {
-            url = Client4.getBaseRoute() + '/image?url=' + encodeURIComponent(url);
-        }
-
+        const url = this.imagesAreProxied ? this.proxyEncodeImageUrl(this.props.link) : this.props.link;
```

Comment:
refactored to make easier to read

Reply:
No needed action right now, but in the future we should keep refactoring to a minimum when fixing anything going into the release branch. The goal should be as little code change as possible to fix the issue and any nice-to-have but not needed refactoring should have a ticket filed for it (or just be done in a separate PR to master)

The reason for this is that more code change means a higher chance of introducing a new bug and new bugs being introduced this late in the release cycle is very undesirable